src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -3206,5 +3206,118 @@ void AnotherLocal()
             }
             EOF();
         }
+
+        [Fact]
+        public void ReturningPartialType_InMethod()
+        {
+            UsingTree("""
+                class C
+                {
+                    void M()
+                    {
+                        partial F() => null;
+                    }
+                }
+                """,
+                // (4,6): error CS1513: } expected
```

Comment:
I guess this matches our behavior with other "non-local" modifiers like `public`?